csharp/src/Apache.Arrow.Adbc/Extensions/StandardSchemaExtensions.cs

```diff
@@ -65,9 +65,29 @@ public static IReadOnlyList<IArrowArray> Validate(this IReadOnlyList<Field> sche
                 else if (field.DataType.TypeId == ArrowTypeId.List)
                 {
                     ListType listType = (ListType)field.DataType;
+
                     if (listType.Fields.Count > 0)
                     {
-                        Validate(listType.Fields, dataField.Children.Select(e => new ContainerArray(e)).ToList());
+                        List<Field> fieldsToValidate = new List<Field>();
+                        List<ContainerArray> arrayDataToValidate = new List<ContainerArray>();
+
+                        for (int j=0;j<listType.Fields.Count;j++)
```

Comment:
Valid list types have just one field, so this loop shouldn't be needed. You can access the item' field via ListType.ValueField.